File: Python/Jump Game.py

class Solution:
    # @param A, a list of integers
    # @return a boolean
    def canJump(self, A):
        length = len(A)
        i = 0
        reach = 0
        while  i< length and i<=reach:
            reach = max(i + A[i], reach)
            print i, reach, A
            i += 1
            
        return i == length

File: Java/src/Jump Game.java
public class Solution
{
  public boolean canJump(int[] A)
  {
    int length = A.length;
    int i = 0;
    int reach = 0;
    while (i < length && i <= reach)
    {
      reach = Math.max(i + A[i], reach);
      i++;
    }

    return i == length;

  }
}
